modules/benchmark/R/read.settings.RR.R

```diff
@@ -0,0 +1,52 @@
+##-------------------------------------------------------------------------------------------------#
+##' For each benchmark entry in a (multi)settings object, get run settings using reference run id 
+##' and add to the settings object
+##'  
+##' @name read.settings.RR
+##' @title Read settings from database using reference run id
+##' @param settings settings or multisettings object
+##' @export 
+##' 
+##' @author Betsy Cowdery 
+
+
+read.settings.RR <- function(settings){
+  library(PEcAn.DB)
+  library(dplyr)
+  
+  # don't know how to check inputs
+  if (is.null(settings$database$bety)) {
+    logger.info("No databasse connection, can't get run information.")
+    return (settings)
+  }
+  
+  bety <- src_postgres(dbname   = settings$database$bety$dbname,
+                       host     = settings$database$bety$host,
+                       user     = settings$database$bety$user,
+                       password = settings$database$bety$password)
+  
+  settings <- tbl(bety,"reference_runs") %>% 
+    filter(id == settings$benchmark$reference_run_id) %>% 
+      dplyr::select(settings) %>% collect() %>% unlist() %>%
+      xmlToList(.,"pecan") %>% append(settings,.) %>% Settings()
```

Comment:
Impressive that you can do the query and insert all in one line. Probably would have taken me a page of code.